internal/rest/avi_evh_obj.go

```diff
@@ -69,6 +70,7 @@ func (rest *RestOperations) RestOperationForEvh(vsName string, namespace string,
 		sslkey_cert_delete, rest_ops = rest.SSLKeyCertCU(aviVsNode.SSLKeyCertRefs, sslkey_cert_delete, namespace, rest_ops, key)
 		pools_to_delete, rest_ops = rest.PoolCU(aviVsNode.PoolRefs, vs_cache_obj, namespace, rest_ops, key)
 		pgs_to_delete, rest_ops = rest.PoolGroupCU(aviVsNode.PoolGroupRefs, vs_cache_obj, namespace, rest_ops, key)
+		string_groups_to_delete, rest_ops = rest.StringGroupVsCU(aviVsNode.StringGroupRefs, vs_cache_obj, namespace, rest_ops, key)
```

Comment:
should this deletion not happen after HTTPPolicy deletion?

Reply:
We are just computing the string groups to delete here. At the time of creating rest ops for deletion, httppolicy delete is rest op is added first. See line 131.

Reply:
No. This function `StringGroupVsCU` will modify rest_ops list which is being sent to `ExecuteRestAndPopulateCache` at line 87

Reply:
**StringGroupVsCU** only adds put and post ops to rest_ops which should be execute first i.e., stringgroup should be created first before it is referred in httppolicyset. **string_groups_to_delete**  are basically the stringgroups that need be deleted and those are deleted in line 131 in correct order after HTTPPolicyDelete.